src/plugins/data/public/search/session_service.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import uuid from 'uuid';
+import {
+  ISessionService,
+  IEsSearchRequest,
+  createRequestHash,
+  SearchStatus,
+  SessionStatus,
+} from '../../common/search';
+import { SearchTimeoutError } from '.';
+
+interface RequestInfo {
+  searchId?: string;
+  status: SearchStatus;
+}
+
+interface SessionInfo {
+  sessionId: string;
+  requests: Record<string, RequestInfo>;
+  status: SessionStatus;
+}
+export class SessionService implements ISessionService {
+  private activeSession?: SessionInfo;
+
+  constructor() {
+    this.start();
```

Comment:
This is similar to my comment about `clear()` inside data plugin. 
Should we ask apps to call it explicitly?